src/component/pages/main-page/main-page.tsx

```diff
@@ -1,33 +1,52 @@
-import { FC } from "react";
+import { FC, useReducer } from "react";
 import { Navbar } from "../../organism/navbar/navbar";
-import { NavbarMenuProvider } from "../../../context/navbar-menu-context";
 import BoxSliderMenu from "../../organism/box-slider-menu/box-slider-menu";
-import { FooterProvider } from "../../../context/footer-context";
 import Footer from "../../organism/footer/footer";
-import { Grid, ThemeProvider } from "@mui/material";
+import { Grid, ThemeProvider, useMediaQuery } from "@mui/material";
 import { theme } from "../../../context/theme-context";
 import { Outlet, useLoaderData } from "react-router";
 import { CategoriesModel } from "../../../api/api-get-categories";
+import {
+  REDUCER_ACTION_TYPE,
+  initState,
+  reducer,
+} from "../../../reducer/navbar-reducer";
 
 export const MainPage: FC = () => {
   const category = useLoaderData() as CategoriesModel[];
+  const breakpointView: boolean = useMediaQuery("(min-width: 768px)");
+  const [state, dispatch] = useReducer(reducer, initState);
 
+  const updateBurgerButton = (): void => {
+    dispatch({
+      type: REDUCER_ACTION_TYPE.HANDLE_BURGER_MENU,
+      payload: state.burgerToggle,
```

Comment:
Can't this state be managed by the reducer, we just have to provide the new value by payload?

Reply:
i don't understand.